\documentclass{article}
\usepackage{color}
\definecolor{BiocBlue}{RGB}{24,129,194}
\usepackage[
  colorlinks, linkcolor=BiocBlue, citecolor=BiocBlue,
  anchorcolor=BiocBlue, urlcolor=BiocBlue
]{hyperref}
\usepackage{todonotes}
\usepackage[margin=1in]{geometry}
\usepackage[
  maxbibnames=99,backend=biber,citestyle=authoryear,sorting=ydnt
]{biblatex}
\addbibresource{citations.bib}
\title{\cRomwell: Run Workflow Description Language (WDL) workflows from R}
\author{Sean R. Davis\thanks{seandavi@gmail.com, to whom correspondence should be addressed}  \\
        Center for Cancer Research\\
        National Cancer Institute\\
        National Institutes of Health\\
        }

\date{\today}

\def\code#1{\texttt{#1}}

\newcommand{\Bioconductor}{\emph{Bioconductor}}
\newcommand{\R}{\emph{R}}
\newcommand\Biocpkg[1]{
  {\href{http://bioconductor.org/packages/#1}%
  {\textit{#1}}}}
\newcommand\Rpkg[1]{
  {\href{https://cran.r-project.org/package=#1}%
  {\textit{#1}}}}
\newcommand\github[2]{
  {\href{http://github.org/#1/#2}%
  {\textit{#2}}}}
\newcommand{\cRomwell}{\github{seandavi}{cRomwell}}

\begin{document}
\maketitle

\begin{abstract}
  Biomedical research and precision medicine are now data-intensive
  endeavors. Complex computational pipelines composed of multiple
  dependent and interrelated tasks are often intrinsic to the
  interpretation of multiple -omics assays, medical imaging, and
  numerous sources of textual and unstructured data. Numerous toolsets
  and workflow frameworks facilitate the design and execution of these
  computational pipelines. Many of these workflow approaches isolate
  the pipeline and workflow descriptions and their execution from
  commonly-used data science tools such as \R that facilitate
  contextual documentation, literate programming, and interactive
  analysis and visualization. The \cRomwell package provides the
  interface between the \R interactive data science environment and
  arbitrary-scale orchestration of complex complex workflows described
  using the Workflow Description Language (WDL). \\
  \textbf{Availability}: \url{https://github.com/seandavi/cRomwell}\\
  \textbf{Keywords}: bioinformatics, genomics, cloud computing,
  reproducible research
\end{abstract}

\section*{Introduction}

Biomedical research has rapidly become a data-intensive science. Many
scientific questions, such as those involving large-scale sequencing,
involve orchestrating computationally complex and demanding data
processing pipelines that transform the raw data into more compact and
information-rich processed data. A large ecosystem of workflow systems
exists (see,
\href{https://github.com/pditommaso/awesome-pipeline}{here} for a
large list). The particular implementation details, strengths and
weaknesses, and suitability applications varies quite
significantly. Ideally, a robust and reliable workflow management
system will address three key challenges (\cite{rabix}): flexibility,
portability, and reproducibility. A flexible workflow management
system will support workflow descriptions and supporting toolkit(s)
that ``gracefully handle large volumes of data with multiple formats''
(\cite{rabix}) and provide generalizations that promote descriptions
that can be applied to different datasets without requiring different
versions of the workflow. Portability, or the ability to run workflows
in multiple compute environments, enabling researchers to move or
expand computational workloads with minimal re-engineering or workflow
development (\cite{rabix}). Portable workflows also significantly
reduce barriers to collaboration and workflow component
re-use. Finally, a workflow management system should support
reproducible computational research approaches (\cite{Sandve2013}).

The Workflow Description Language (WDL) is ``a new workflow definition
language that is designed from the ground up as a human-readable and
-writable way to express tasks and workflows.'' \cite{WDLsite}). A WDL
workflow describes the inputs, outputs, and transformations between
them in a text format defined by a well-defined specification. The
specification allows relatively straightforward definition of
computational tasks, each composed of a command-line script,
associated inputs and outputs, and compute resource requirements
necessary to run the task. WDL tasks can be combined with other
dependent tasks to form complex and parallelizable workflows.

A key component of a successful workflow man

\printbibliography

  


\end{document}
