%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
% Welcome to Overleaf --- just edit your article on the left,
% and we'll compile it for you on the right. If you give 
% someone the link to this page, they can edit at the same
% time. See the help menu above for more info. Enjoy!
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
% For more detailed article preparation guidelines, please see:
% https://f1000research.com/for-authors/article-guidelines/software-tool-articles and http://f1000research.com/data-preparation

\documentclass[9pt,a4paper]{extarticle}
\usepackage{f1000_styles}
\usepackage{hyperref}

%% Default: numerical citations
%\usepackage[numbers]{natbib}

%% Uncomment this lines for superscript citations instead
% \usepackage[super]{natbib}

%% Uncomment these lines for author-year citations instead
% \usepackage[round]{natbib}
% \let\cite\citep

\begin{document}
\pagestyle{front}

\title{cRomwell: a R Interface to an advanced workflow management system.}
\author[1]{Sean R. Davis}
\affil[1]{
  \href{mailto:seandavi@gmail.com}{seandavi@gmail.com}, 
  \href{http://orcid.org/0000-0002-8991-6458}{ORCID}, 
  to whom correspondence should be addressed\\
  Center for Cancer Research, National Cancer Institute, 
  National Institutes of Health, 
  37 Convent Drive, Bethesda, MD, 20892, USA}

\maketitle
\thispagestyle{front}

Please list all authors that played a significant role in developing the software tool and/or writing the article. Please provide full affiliation information (including full institutional address, ZIP code and e-mail address) for all authors, and identify who is/are the corresponding author(s).

\begin{abstract}
  Biomedical research and precision medicine are now data-intensive
  endeavors. Complex computational pipelines composed of multiple
  dependent and interrelated tasks are now intrinsic to the
  interpretation of multiple -omics assays, medical imaging, and
  numerous sources of textual and unstructured data. Numerous toolsets
  and workflow frameworks facilitate the design and execution of these
  computational pipelines. Many of these workflow approaches divorce
  the pipeline and workflow descriptions and their execution from
  commonly-used data science tools such as R that facilitate
  contextual documentation, literate programming, and interactive
  analysis and visualization. The cRomwell package described here
  provides the interface between the R interactive data science
  environment and arbitrary-scale orchestration of complex complex
  workflows described using the Workflow Description Language (WDL).
\end{abstract}

\section*{Keywords}

bioinformatics, genomics, cloud computing, reproducible research

\cite{Smith:2012qr}

\clearpage
\pagestyle{main}
\section*{Introduction}

Biomedical research has rapidly become a data-intensive science. Many scientific questions, such as those involving large-scale sequencing, involve orchestrating computationally complex and demanding data processing pipelines that transform the raw data into more compact and information-rich processed data. A large ecosystem of workflow systems exists (see, \href{https://github.com/pditommaso/awesome-pipeline}{here} for a large list). The particular implementation details, strengths and weaknesses, and suitability applications varies quite significantly. However, a robust and reliable workflow management system will address three key challenges: flexibility, portability, and reproducibility \cite{Smith:2012qr}.


Additional 

\section*{Methods}
\subsection*{Implementation}
For software tool papers, this section should address how the tool works and any relevant technical details required for implementation of the tool by other developers.  

\subsection*{Operation}
This part of the methods should include the minimal system requirements needed to run the software and an overview of the workflow for the tool for users of the tool.


\section*{Results} % Optional - only if novel data or analyses are included
This section is only required if the paper includes novel data or analyses, and should be written as a traditional results section.

\section*{Use Cases} % Optional - only if NO new datasets are included
This section is required if the paper does not include novel data or analyses. 
Examples of input and output files should be provided with some explanatory context.  Any novel or complex variable parameters should also be explained in sufficient detail to allow users to understand and use the tool's functionality.

\section*{Discussion} % Optional - only if novel data or analyses are included
This section is only required if the paper includes novel data or analyses, and should be written in the same style as a traditional discussion section.
Please include a brief discussion of allowances made (if any) for controlling bias or unwanted sources of variability, and the limitations of any novel datasets.


\section*{Conclusions} % Optional - only if novel data or analyses are included
This section is only required if the paper includes novel data or analyses, and should be written as a traditional conclusion.

\section*{Summary} % Optional - only if NO new datasets are included
This section is required if the paper does not include novel data or analyses.  It allows authors to briefly summarize the key points from the article.

\section*{Data availability} % Optional - only if novel data or analyses are included
Please add details of where any datasets that are mentioned in the paper, and that have not have not previously been formally published, can be found.  If previously published datasets are mentioned, these should be cited in the references, as per usual scholarly conventions.

\section*{Software availability}
This section will be generated by the Editorial Office before publication. Authors are asked to provide some initial information to assist the Editorial Office, as detailed below.
\begin{enumerate}
\item URL \url{https://github.com/seandavi/cRomwell}
\item Link to source code as at time of publication ({\textit{F1000Research}} TO GENERATE)
\item Link to archived source code as at time of publication ({\textit{F1000Research}} TO GENERATE)
\item Software license (AUTHOR TO PROVIDE; required)
\end{enumerate}



\section*{Author contributions}
In order to give appropriate credit to each author of an article, the individual
contributions of each author to the manuscript should be detailed in this section. We
recommend using author initials and then stating briefly how they contributed.

\section*{Competing interests}
All financial, personal, or professional competing interests for any of the authors that
could be construed to unduly influence the content of the article must be disclosed and will be displayed alongside the article. If there are no relevant competing interests to declare, please add the following: 'No competing interests were disclosed'.

\section*{Grant information}
Please state who funded the work discussed in this article, whether it is your employer, a grant funder etc. Please do not list funding that you have that is not relevant to this
specific piece of research. For each funder, please state the funder’s name, the grant
number where applicable, and the individual to whom the grant was assigned.
If your work was not funded by any grants, please include the line: ‘The author(s)
declared that no grants were involved in supporting this work.’

\section*{Acknowledgments}
This section should acknowledge anyone who contributed to the research or the
article but who does not qualify as an author based on the criteria provided earlier
(e.g. someone or an organization that provided writing assistance). Please state how
they contributed; authors should obtain permission to acknowledge from all those
mentioned in the Acknowledgments section.

Please do not list grant funding in this section.

\bibliographystyle{unsrtnat}
\bibliography{sample}

\bigskip
References can be listed in any standard referencing style and should be consistent between references within a given article.

Reference management systems such as Zotero provide options for exporting bibliographies as Bib\TeX{} files. Bib\TeX{} is a bibliographic tool that is used with \LaTeX{} to help organize the user's references and create a bibliography. This template contains an example of such a file, \texttt{sample.bib}, which can be replaced with your own. Use the \verb|\cite| command  to create in-text citations, like this and this .


\section*{USING LATEX}
Some examples of commonly used \LaTeX{}  commands and features are listed below, to help you get started.


\subsection*{Sections}

Use section and subsection commands to organize your document. \LaTeX{} handles all the formatting and numbering automatically. Use ref and label commands for cross-references.


\subsection*{Tables}

Use the table and tabledata commands for basic tables --- see Table~\ref{tab:widgets}, for example.
\begin{table}[h!]
\hrule \vspace{0.1cm}
\caption{\label{tab:widgets}An example of a simple table with caption.}
\centering
\begin{tabledata}{llr} 
\header First name & Last Name & Grade \\ 
\row John & Doe & $7.5$ \\ 
\row Richard & Miles & $2$ \\ 
\end{tabledata}
\end{table}

\subsection*{Figures}
You can upload a figure (JPEG, PNG or PDF) using the files menu. To include it in your document, use the includegraphics command (see the example below in the source code).

Please give figures appropriate filenames eg: figure1.pdf, figure2.png.

Figure legends should briefly describe the key messages of the figure such that the figure can stand alone from the main text. However, all figures should also be discussed in the article text. Each legend should have a concise title of no more than 15 words. The legend itself should be succinct, while still explaining all symbols and abbreviations. Avoid lengthy descriptions of methods.

For any figures reproduced from another publication (as long as appropriate permission has been obtained from the copyright holder —see under the heading 'Submission'), please include a line in the legend to state that: 'This figure has been reproduced with kind permission from [include original publication citation]'.

\begin{figure}
\centering
\includegraphics[width=0.4\textwidth]{frog.jpg}
\caption{\label{fig:your-figure}Your figure legend goes here; it should be succinct, while still explaining all symbols and abbreviations. }
\end{figure}



\subsection*{Mathematics}

\LaTeX{} is great at typesetting mathematics. Let $X_1, X_2, \ldots, X_n$ be a sequence of independent and identically distributed random variables with $\text{E}[X_i] = \mu$ and $\text{Var}[X_i] = \sigma^2 < \infty$, and let
$$S_n = \frac{X_1 + X_2 + \cdots + X_n}{n}
      = \frac{1}{n}\sum_{i}^{n} X_i$$
denote their mean. Then as $n$ approaches infinity, the random variables $\sqrt{n}(S_n - \mu)$ converge in distribution to a normal $\mathcal{N}(0, \sigma^2)$.

\subsection*{Lists}

You can make lists with automatic numbering \dots

\begin{enumerate}
\item Like this,
\item and like this.
\end{enumerate}
\dots or bullet points \dots
\begin{itemize}
\item Like this,
\item and like this.
\end{itemize}

% See this guide for more information on BibTeX:
% http://libguides.mit.edu/content.php?pid=55482&sid=406343

% For more author guidance please see:
% https://f1000research.com/for-authors/article-guidelines/software-tool-articles


% When all authors are happy with the paper, use the 
% ‘Submit to F1000RESEARCH' button from the menu above
% to submit directly to the open life science journal F1000Research.

% Please note that this template results in a draft pre-submission PDF document.
% Articles will be professionally typeset when accepted for publication.

% We hope you find the F1000Research Overleaf template useful,
% please let us know if you have any feedback using the help menu above.


\end{document}
